alternatives.c

```diff
@@ -508,10 +517,13 @@ static int removeLinks(struct linkSet *l, const char *altDir, int flags) {
     }
     if (FL_TEST(flags)) {
         printf(_("would remove %s\n"), l->facility);
-    } else if (isLink(l->facility) && unlink(l->facility) && errno != ENOENT) {
-        fprintf(stderr, _("failed to remove link %s: %s\n"), l->facility,
+    } else {
+        if (isLink(l->facility) && (!FL_KEEP_FOREIGN(flags) || facilityBelongsToUs(l->facility, altDir)))
```

Comment:
Isn't there missing ``}``. Also, wouldn't be cleaner to make it ``else if`` as before?